### PR TITLE
Added version-check to CI 

### PR DESCRIPTION
**Pull request by dheerajd5**
This PR adds a version-check to CI checks. It takes the version fetched from git and from the file `version.h` and compares it.
#497

### CODE REVIEW

**Comment by gavv** [comment]
Thanks for PR! Will review it in upcoming days.

File: scripts/ci_checks/linux-checks/check-version.sh [hunk 1]
```diff
@@ -0,0 +1,20 @@
...
+tagname=${tagname: -5}
+header_tagname=$(python3 scripts/scons_helpers/parse-version.py)
+
+if [[ "$tagname" != "$header_tagname" ]]; then
+      echo
+      echo "version check FAILED"
+      echo "the git tag name and version declared in version.h is not the same"
+      echo "${tagname} git version tag"
+      echo "${header_tagname} header version tag"
+      echo
```
Comment (adrianopol) [inline scripts/ci_checks/linux-checks/check-version.sh:16]: `echo >&2` is preferred

File: scripts/ci_checks/linux-checks/check-version.sh [hunk 2]
```diff
@@ -0,0 +1,20 @@
+#!/usr/bin/env bash
+
+set -euo pipefail
+
+
+tagname=$(git tag -l)
+tagname=${tagname: -5}
```
Comment (adrianopol) [inline scripts/ci_checks/linux-checks/check-version.sh:7]: very fragile; suggestion:

```suggestion
tagname="$(git tag -l | tail -n1)"
```

File: scripts/ci_checks/linux-checks/check-version.sh [hunk 3]
```diff
@@ -0,0 +1,20 @@
...
+header_tagname=$(python3 scripts/scons_helpers/parse-version.py)
```
Comment (adrianopol) [inline scripts/ci_checks/linux-checks/check-version.sh:8]: more secure way:

```suggestion
header_tagname="$(python3 scripts/scons_helpers/parse-version.py)"
```

File: scripts/ci_checks/linux-checks/check-version.sh [hunk 4]
```diff
@@ -0,0 +1,20 @@
...
+
+if [[ "$tagname" != "$header_tagname" ]]; then
+      echo
```
Comment (adrianopol) [inline scripts/ci_checks/linux-checks/check-version.sh:11]: nit: common indentation size in shell-scripts is 2 spaces

File: scripts/ci_checks/linux-checks/check-version.sh [hunk 5]
```diff
@@ -0,0 +1,20 @@
...
+      echo "version check FAILED"
+      echo "the git tag name and version declared in version.h is not the same"
+      echo "${tagname} git version tag"
+      echo "${header_tagname} header version tag"
+      echo
+      exit 1
+  else
```
Comment (adrianopol) [inline scripts/ci_checks/linux-checks/check-version.sh:18]:
```suggestion
else
```

File: scripts/ci_checks/linux-checks/check-version.sh [hunk 6]
```diff
@@ -0,0 +1,20 @@
...
+     echo "version check OK"
```
Comment (adrianopol) [inline scripts/ci_checks/linux-checks/check-version.sh:19]: nit:
```suggestion
     echo "version check SUCCEEDED"
```

File: scripts/ci_checks/linux-checks/check-version.sh (hunk 5 above)
Comment (adrianopol) [reply scripts/ci_checks/linux-checks/check-version.sh:18]: BTW, else is useless here (as soon as you do `exit` in the first branch)

File: scripts/ci_checks/linux-checks/check-version.sh (hunk 1 above)
Comment (dheerajd5) [reply scripts/ci_checks/linux-checks/check-version.sh:15]: Hey so in the other CI check scripts, such as `check-formatting` , `echo` is used, even for error messages, should I stick with that or make the change to `echo >&2` ?

**Comment by dheerajd5** [comment]
Hey, I rebased over existing commits by mistake, I'll put up a new PR with the code reviewed changes.

File: scripts/ci_checks/linux-checks/check-version.sh (hunk 1 above)
Comment (adrianopol) [reply scripts/ci_checks/linux-checks/check-version.sh]: @dheerajd5 , hello, this behavior is preferred, yes, as soon as it doesn't break the script's functionality.

**Comment by adrianopol** [comment]
Hi, there is no need to close this one, you may just fix your branch locally (e.g. rebase on `roc-streaming:develop` branch) and then make `git push -f`.

**Comment by dheerajd5** [comment]
Oh, sorry I'm new to this, I'll keep it in mind next time.

**Comment by adrianopol** [comment]
no problem, thank you!

**Pull request closed without merging**